### PR TITLE
chore: add labels to docker images

### PR DESCRIPTION
**Pull request by candrews**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

Use docker/metadata-action@v4 to add labels to the docker images.

Fixes: https://github.com/bridgecrewio/checkov/issues/4379

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -270,6 +270,12 @@ jobs:
           registry: ghcr.io
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
+      -
+        name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4.3.0
```
Comment (gruebel) [inline .github/workflows/build.yml:276]: please use the commit hash and add the version as a comment behind it.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -270,6 +270,12 @@ jobs:
...
+        with:
+          tag-sha: true # add git short SHA as Docker tag
```
Comment (gruebel) [inline .github/workflows/build.yml:278]: I didn't find this input field, only `tags`

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -270,6 +270,12 @@ jobs:
           registry: ghcr.io
           username: ${{ github.actor }}
           password: ${{ secrets.GITHUB_TOKEN }}
+      -
+        name: Docker meta
```
Comment (gruebel) [inline .github/workflows/build.yml:274]:
```suggestion
      - name: Docker meta
```

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -270,6 +270,12 @@ jobs:
...
+        id: meta
```
Comment (gruebel) [inline .github/workflows/build.yml:275]:
```suggestion
        id: docker_meta
```

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -278,6 +284,7 @@ jobs:
           context: .
           no-cache: true
           load: true
+          labels: ${{ steps.meta.outputs.labels }}
```
Comment (gruebel) [inline .github/workflows/build.yml:287]:
```suggestion
          labels: ${{ steps.docker_meta.outputs.labels }}
```

**Review by gruebel — COMMENTED** [review]
thanks for the contribution 💪 I added a couple of comments

File: .github/workflows/build.yml (hunk 2 above)
Comment (candrews) [reply .github/workflows/build.yml:278]: You're of course right, the documentation confused me a bit. I've now removed `tag-sha`.

File: .github/workflows/build.yml (hunk 4 above)
Comment (candrews) [reply .github/workflows/build.yml:275]: I've made this change

File: .github/workflows/build.yml (hunk 5 above)
Comment (candrews) [reply .github/workflows/build.yml:287]: I've made this change

File: .github/workflows/build.yml (hunk 3 above)
Comment (candrews) [reply .github/workflows/build.yml:274]: I've made this change

File: .github/workflows/build.yml (hunk 1 above)
Comment (candrews) [reply .github/workflows/build.yml:276]: I've made this change

**Review by gruebel — APPROVED** [review]
thanks for the contribution, let's see how it works tomorrow 🚀

**Review by Eliran-Turgeman — APPROVED** [review]

**Pull request merged**